jwst/datamodels/wcs_ref_models.py

```diff
@@ -341,6 +341,56 @@ def to_fits(self):
         raise NotImplementedError("FITS format is not supported for this file.")
 
 
+class NirissSOSSTableModel(ReferenceFileModel):
+    """
+    A model for a reference file of type "specwcs" for NIRISS SOSS wavelength tables.
+
+    Parameters
+    ----------
+    models: `~astropy.modeling.Tabular`
+          NIRISS SOSS wavelength  model
+    """
+    schema_url = "http://stsci.edu/schemas/jwst_datamodel/specwcs_niriss_soss.schema"
+    reftype = "specwcs"
+
+    def __init__(self, init=None,
+                 models=None,
+                 **kwargs):
+        super().__init__(init=init, **kwargs)
+
+        if init is None:
+            self.populate_meta()
+        if models is not None:
+            self.models = models
+        
+    def populate_meta(self):
+        self.meta.instrument.name = "NIRISS"
+        self.meta.instrument.detector = "NIS"
+        self.meta.exposure.type = "NIS_SOSS"
+        self.meta.reftype = self.reftype
+
+    def validate(self):
+        super(NirissSOSSTableModel, self).validate()
+        try:
+            assert isinstance(self.meta.input_units, (str, u.NamedUnit))
+            assert isinstance(self.meta.output_units, (str, u.NamedUnit))
+            assert self.meta.instrument.name == "NIRISS"
```

Comment:
Given that all of these values are hardwired just 10 lines above, is it even possible for the asserts to ever fail and hence are they really useful?